src/tests/profiler/common/ProfilerTestRunner.cs

```diff
@@ -83,17 +83,20 @@ public class ProfilerTestRunner
             process.BeginOutputReadLine();
 
             process.WaitForExit();
-            if (process.ExitCode == 100 && verifier.HasPassingOutput)
-            {
-                return 100;
-            }
-            else
+
+            if (!verifier.HasPassingOutput)
             {
                 LogTestFailure("Profiler tests are expected to contain the text \'" + verifier.SuccessPhrase + "\' in the console output " +
                     "of the profilee app to indicate a passing test. Usually it is printed from the Shutdown() method of the profiler implementation. This " +
                     "text was not found in the output above.");
-                return process.ExitCode == 100 ? process.ExitCode : -1;
             }
+
+            if (process.ExitCode != 100)
+            {
+                LogTestFailure($"Profilee returned exit code {process.ExitCode} instead of expected exit code 100.");
+            }
+
+            return 100;
```

Comment:
I think you said in an earlier PR that the exit code didn't matter for profiler tests.  It still would be nice to have a comment to that effect.

Reply:
I don't remember exactly what I said, but I think I was trying to say that the exit code doesn't matter for that particular test. We do check the exit code, it's just that a lot of profiler tests don't use it to indicate failure. Some of them do, however, and crashes will return a non 100 exit code.

Reply:
I guess I just wanted to make it clear to someone other than you that `return 100;` can happen even though the test failed.  I just feels confusing to me.  If we hadn't talked about it, I would have asked again.

Reply:
Sure, I added a comment

Reply:
I agree that logging a failure message and then returning 100 feels counter intuitive. If the exit code matters I was expecting there would be some conditional logic or if it doesn't matter then perhaps we should return 0?

Reply:
I misunderstood what Steve was saying. As Noah points out in the edit above LogTestFailure intentionally throws an unhandled exception to fail the test. I can see how that is unclear to others though, since it just says log in the name

Reply:
Precisely **unclear**

Reply:
I just changed it from LogTestFailure to FailFastWithMessage so hopefully its more clear just from the name

Reply:
Thanks